easybuild/tools/job/gc3pie.py

```diff
@@ -197,7 +197,9 @@ def complete(self):
         self._engine.add(self.jobs)
 
         # in case you want to select a specific resource, call
-        self._engine.select_resource(build_option('job_target_resource'))
+        target_resource = build_option('job_target_resource')
+        if target_resource:
+            self._engine.select_resource(target_resource)
```

Comment:
@riccardomurri: The `select_resource` method currently doesn't return the number of *enabled* resources, like you said.

It does after applying this patch to GC3Pie:

```diff
--- gc3libs/core.py	(revision 4221)
+++ gc3libs/core.py	(working copy)
@@ -163,6 +163,7 @@
           - or it can be a string: only resources whose name matches
             (wildcards ``*`` and ``?`` are allowed) are retained.
         """
+        enabled = 0
         for lrms in self.resources.itervalues():
             try:
                 if not match(lrms):
@@ -170,7 +171,9 @@
             except:
                 if not fnmatch(lrms.name, match):
                     lrms.enabled = False
-        return len(self.resources)
+            if lrms.enabled:
+                enabled += 1
+        return enabled
 
     def free(self, app, **extra_args):
         """
```